packages/rocketchat-search-lib/lib/SearchProviderRegistry.js

```diff
@@ -1,13 +1,33 @@
+import {SearchProvider} from './SearchProvider';
+
 export class SearchProviderRegistry {
 	constructor() {
+		this.providers = {};
+		this.activeProviderIdentifier = '';
 	}
+
 	/**
 	 * Registers a search provider which can potentially provide full-text-search
 	 * @param searchProvider
+	 * @return {boolean} whether the provider has been registered successfully
 	 */
-	add(searchProvider){
+	add(searchProvider) {
+		if (!(searchProvider instanceof SearchProvider)) {
+			throw new Error('Invalid provider object, it must extend "SearchProvider"');
+		}
+
+		if (this.providers[searchProvider.identifier]) {
+			return false;
+		}
+
+		this.providers[searchProvider.identifier] = searchProvider;
```

Comment:
Since the documentation on this method says it returns a boolean, make sure you return one at the end of it. :)

Reply:
Of course. Thx. There's basically no implementation yet. Intention of sharing it was to be able to discuss about the interfaces. I might add a bit more during the next days.